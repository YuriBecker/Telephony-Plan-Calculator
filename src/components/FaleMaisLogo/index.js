import React from 'react';
import { Text } from '@ui-kitten/components';
import { View, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { PhoneIcon } from '../Icons/Icons';
import colors from '../../styles/colors';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  text: {
    color: colors.primary,
    fontWeight: 'bold',
  },
});

const index = ({ titleCategory, iconWidth, iconHeight }) => {
  return (
    <View style={styles.container}>
      <Text category={titleCategory} style={styles.text}>
        FaleMais
      </Text>
      <PhoneIcon
        fill={colors.primary}
        width={iconWidth}
        height={iconHeight}
        style={{ marginTop: 5, marginLeft: 5 }}
      />
    </View>
  );
};

export default index;

index.propTypes = {
  titleCategory: PropTypes.string,
  iconWidth: PropTypes.number,
  iconHeight: PropTypes.number,
};

index.defaultProps = {
  titleCategory: 'h1',
  iconWidth: 38,
  iconHeight: 38,
};
