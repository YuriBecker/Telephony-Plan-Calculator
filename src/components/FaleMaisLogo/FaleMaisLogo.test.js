/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow } from 'enzyme';

import FaleMaisLogo from './index';

describe('Testing FaleMaisLogo component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FaleMaisLogo />);
  });

  it('should render correctly - Default', () => {
    const component = shallow(<FaleMaisLogo />);
    expect(component).toMatchSnapshot('Default  snapshot');
  });

  it('should render correctly - Diferent icon size', () => {
    const component = shallow(<FaleMaisLogo iconHeight={24} iconWidth={24} />);
    expect(component).toMatchSnapshot('Diferent icon size snapshot');
  });

  it('should render correctly - Diferent text category', () => {
    const component = shallow(<FaleMaisLogo titleCategory="h2" />);
    expect(component).toMatchSnapshot('Diferent text category');
  });
});
