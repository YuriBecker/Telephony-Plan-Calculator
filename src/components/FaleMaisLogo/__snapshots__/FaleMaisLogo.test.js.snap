// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FaleMaisLogo component should render correctly - Default: Default  snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <TextComponent
    category="h1"
    style={
      Object {
        "color": "#1db954",
        "fontWeight": "bold",
      }
    }
  >
    FaleMais
  </TextComponent>
  <PhoneIcon
    fill="#1db954"
    height={38}
    style={
      Object {
        "marginLeft": 5,
        "marginTop": 5,
      }
    }
    width={38}
  />
</View>
`;

exports[`Testing FaleMaisLogo component should render correctly - Diferent icon size: Diferent icon size snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <TextComponent
    category="h1"
    style={
      Object {
        "color": "#1db954",
        "fontWeight": "bold",
      }
    }
  >
    FaleMais
  </TextComponent>
  <PhoneIcon
    fill="#1db954"
    height={24}
    style={
      Object {
        "marginLeft": 5,
        "marginTop": 5,
      }
    }
    width={24}
  />
</View>
`;

exports[`Testing FaleMaisLogo component should render correctly - Diferent text category: Diferent text category 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <TextComponent
    category="h2"
    style={
      Object {
        "color": "#1db954",
        "fontWeight": "bold",
      }
    }
  >
    FaleMais
  </TextComponent>
  <PhoneIcon
    fill="#1db954"
    height={38}
    style={
      Object {
        "marginLeft": 5,
        "marginTop": 5,
      }
    }
    width={38}
  />
</View>
`;
